src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -13,59 +16,103 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class BasicUserService implements UserService {
     private final UserRepository userRepository;
+    private final BinaryContentRepository contentRepository;
+    private final UserStatusRepository userStatusRepository;
 
     @Override
-    public void createUser(User user) {
+    public User createUser(User user) {
         List<User> all = userRepository.findAll();
         for (User findUser : all) {
-            if (user.getUserName().equals(findUser.getUserName())) {
-                log.info("중복된 이름입니다.");
-                return;
+            //동명이인이 있을수있는데 Username이 다른유저와 같으면 안된다는 요구사항이 필요한 요소일까요?
+            if (user.getLoginId().equals(findUser.getLoginId())) {
+                throw new IllegalArgumentException("중복된 아이디입니다.");
             }
             if (user.getUserEmail().equals(findUser.getUserEmail())) {
-                log.info("중복된 이메일입니다.");
-                return;
+                throw new IllegalArgumentException("중복된 이메일 입니다.");
             }
         }
-        userRepository.createUser(user.getId(), user);
-        UserStatus userStatus = new UserStatus(user.getId());
+        if (user.getAttachProfile() != null) {
+           contentRepository.save(user.getAttachProfile());
+        }
+        User savedUser = userRepository.createUser(user);
+        UserStatus userStatus = new UserStatus(savedUser.getId());
+        userStatusRepository.save(userStatus);
+        return savedUser;
     }
 
     @Override
     public Optional<CheckUserDto> findUser(UUID id) {
-        return userRepository.findById(id).map(CheckUserDto::new);
+        User user = userRepository.findById(id).orElseThrow(()->new IllegalArgumentException("해당 사용자가 존재하지 않습니다."));
+        return Optional.of(new CheckUserDto(user, isUserOnline(id)));
     }
 
     @Override
     public Optional<CheckUserDto> findByloginId(String loginId) {
-        return userRepository.findByloginId(loginId).map(CheckUserDto::new);
+        User user = userRepository.findByloginId(loginId).orElseThrow(()->new IllegalArgumentException("해당 사용자가 존재하지 않습니다."));
+        return Optional.of(new CheckUserDto(user, isUserOnline(user.getId())));
     }
 
     @Override
-    public List<User> findAllUsers() {
-        return userRepository.findAll();
+    public List<CheckUserDto> findAllUsers() {
+        return userRepository.findAll().stream()
+                .map(user->{ UserStatus userStatus = userStatusRepository.findByUserId(user.getId()).orElseThrow(()->new IllegalArgumentException("해당 사용자 상태정보를 찾을 수 없습니다."));
+                    return new CheckUserDto(user, isUserOnline(user.getId()));
+                }).collect(Collectors.toList());
```

Comment:
`.collect(Collectors.toList());` 대신 Stream.toList()를 사용하는게 코드가 더 간결하고 immutable할 것 같네요.  
```suggestion
        return userRepository.findAll().stream()
                .map(user -> {
                    userStatusRepository.findByUserId(user.getId())
                            .orElseThrow(() -> new IllegalArgumentException("해당 사용자 상태정보를 찾을 수 없습니다."));
                    return new CheckUserDto(user, isUserOnline(user.getId()));
                })
                .toList();
```